Review

IM-GYURI opened the pull request:
### 변경사항
**AS-IS**
- commerce 프로젝트 내에서 seller 관련 api와 customer api 구현
- seller-api 모듈과 customer-api 모듈 내 중복된 코드 다수

**TO-BE**
- 멀티 모듈 구현
- seller-api 모듈과 customer-api 모듈 내 중복된 코드들을 common 모듈 내로 이동
- Trouble Shooting 작성

IM-GYURI committed:
docs : add Trouble shootings

about MariaDB and Gradle

IM-GYURI committed:
docs : add Trouble shootings

about Multi Module Gradle

IM-GYURI committed:
fix : delete sub gradle files

IM-GYURI committed:
feat : add root directory settings

IM-GYURI committed:
feat : change import

from sellerapi to common

IM-GYURI committed:
feat : change import

from customerapi to common

IM-GYURI committed:
feat : add common module

IM-GYURI committed:
refactore : move CustomAccessDeniedHandler to common module

IM-GYURI committed:
refactore : move CustomAccessDeniedHandler to common module

IM-GYURI committed:
feat : change application.properties to application.yml

IM-GYURI committed:
refactor : change import of Role

from customerapi to common

IM-GYURI committed:
refactor : change import of Role

from customerapi to common

IM-GYURI committed:
refactor : change import of exception and util

from customerapi to common

IM-GYURI committed:
refactor : delete common classes

IM-GYURI committed:
refactor : change import of exception

from customerapi to common

IM-GYURI committed:
refactor : change import of exception

from sellerapi to common

IM-GYURI committed:
refactor : change import of Role

from sellerapi to common

IM-GYURI committed:
refactor : change import of Role

from sellerapi to common

IM-GYURI committed:
refactor : change import of exception and util

from sellerapi to common

IM-GYURI committed:
docs : add Trouble Shootings

about Multi Module problem
'Project can't resolve the Bean'

IM-GYURI committed:
feat : add BeanConfig

for Scanning 'zerobase.common'

IM-GYURI committed:
refactor : move CustomException

from each sub module to common

IM-GYURI committed:
refactor : move ErrorCode

from each sub module to common

IM-GYURI committed:
refactor : move ErrorResponse

from each sub module to common

IM-GYURI committed:
refactor : move GlobalExceptionHandler

from each sub module to common

IM-GYURI committed:
refactor : move KeyGenerator

from each sub module to common

IM-GYURI committed:
refactor : move Role

from each sub module to common

IM-GYURI committed:
docs : check done features

spring-kotlin-lover commented on common/src/main/java/zerobase/common/exception/ErrorCode.java:
```diff
@@ -14,6 +14,10 @@ public enum ErrorCode {
   CUSTOMER_ALREADY_EXISTS(BAD_REQUEST, "이미 존재하는 고객입니다."),
   CUSTOMER_NOT_FOUND(NOT_FOUND, "고객을 찾을 수 없습니다"),
 
+  //Seller
```
전체적으로 모듈 설계를 domain 기반으로 나누어주셨는데, ErrorCode 에 대한 것도 domain 별로 나누어지는 방향이 전체적인 설계 방향성과 일치할듯 싶습니다.

어떻게 하면 도메인 별로 error code 를 각자 정의해 볼 수 있을까요?

IM-GYURI replied:
Error를 처리하는 클래스들은 현재 상태와 같이 공용으로 사용하고, ErrorCode 클래스만 각각의 모듈 아래로 이동시킬 수 있을 것 같습니다. 또는 공통으로 사용되는 global error code들만 common 모듈 하에 두고, customer 관련 / seller 관련 error code들로 구분하여 해당 모듈 아래로 이동시킬 수 있을 것 같습니다.

생각나는 방법은 이렇게 두 가지인데, 더 좋은 방법이 있을까요?

spring-kotlin-lover reviewed (commented):
전반적으로 각 도메인별로 모듈을 설계해보시려는 시도가 좋았네요. 다만 예외쪽도 추가적인 고민이 포인트가 필요하여 개선점에 대해 고민해보고 정리해보면 좋겠네요~!

IM-GYURI committed:
feat : add Pattern of name and phone number

name Pattern = length of 2 ~ 10
phone number Patter = 010-0000-0000

IM-GYURI committed:
refactor : delete space

IM-GYURI committed:
feat : add Pattern of name and phone number

name Pattern = length of 2 ~ 10
phone number Pattern = 010-0000-0000

add message for address

IM-GYURI committed:
refactor : delete space

IM-GYURI committed:
feat : get information myself

IM-GYURI committed:
feat : add properties

phone, address, point, birth, createdAt, modifiedAt

IM-GYURI committed:
feat : add findByEmail

find by email and return CustomerDto

IM-GYURI committed:
feat : add SignatureException Handling

IM-GYURI committed:
refactor : change requestMatchers

from new AntPathRequestMatcher to just path string

IM-GYURI committed:
feat : add patch customer information

IM-GYURI committed:
feat : add update customer

IM-GYURI committed:
feat : add find by customer Key

IM-GYURI committed:
feat : add edit customer information

IM-GYURI committed:
docs : check done things

IM-GYURI committed:
feat : add Dto for Edit

IM-GYURI committed:
feat : add delete customer

IM-GYURI committed:
feat : add delete customer and validation by customerKey

IM-GYURI committed:
feat : add delete by customer key and exists by customer key

dev-sam32 commented:
확인이 늦었습니다! 너무 수고 많으셨습니다!

IM-GYURI committed:
feat : add get information, edit information, delete information of seller

IM-GYURI committed:
feat : add get information, edit information, delete information of seller

IM-GYURI committed:
feat : add properties

IM-GYURI committed:
feat : add updateSeller

IM-GYURI committed:
feat : add existsBySellerKey, findBySellerKey, deleteBySellerKey

IM-GYURI committed:
feat : add findByEmail, edit, delete, validateSellerExistsBySellerKey

IM-GYURI committed:
feat : add Dto for Edit

IM-GYURI committed:
feat : add Category enum

book, clothes, cosmetic, digital, food, furniture, household goods, kitchen utensils, pet goods, sports

IM-GYURI committed:
feat : add Product Entity

id, product key, seller key (FK), name, category, price, stock, description

IM-GYURI committed:
refactor : move check authentication

from controller to service

IM-GYURI committed:
refactor : move check authentication

from controller to service

IM-GYURI committed:
docs : check done things

IM-GYURI committed:
feat : add PreAuthorize

IM-GYURI committed:
refactor : delete AntPathRequestMatcher

IM-GYURI committed:
feat : add builder and setSeller

build ProductEntity
set SellerEntity

IM-GYURI committed:
docs : check done things

IM-GYURI committed:
feat : add findBySellerKey

IM-GYURI committed:
feat : add common EnumValidator

IM-GYURI committed:
feat : add common ValidEnum

IM-GYURI committed:
feat : add product registration

IM-GYURI committed:
feat : add Dto for Product

IM-GYURI committed:
feat : add Product Repository

IM-GYURI committed:
feat : add product registration

IM-GYURI committed:
feat : add Dto for registration

IM-GYURI committed:
feat : add Error Codes for Product

IM-GYURI committed:
feat : add edit product information

IM-GYURI committed:
feat : add update product

IM-GYURI committed:
feat : add find by productKey

IM-GYURI committed:
feat : add edit, findByProductKeyOrThrow, validateAuthorizationAndGetSeller

IM-GYURI committed:
feat : add @Valid

IM-GYURI committed:
feat : add Dto for Edit Product

IM-GYURI committed:
docs : check done things

IM-GYURI committed:
feat : add delete product

IM-GYURI committed:
feat : add deleteByProductKey

IM-GYURI committed:
feat : add delete product

IM-GYURI committed:
feat : change error message

from SELLER NOT FOUND to IS NOT SAME SELLER

IM-GYURI committed:
refactor : change CustomException to CommonCustomException

IM-GYURI committed:
refactor : change CustomException to CustomerCustomException

IM-GYURI committed:
style : delete blank

IM-GYURI committed:
refactor : change ErrorCode to CustomerErrorCode

IM-GYURI committed:
refactor : change from class to interface

IM-GYURI committed:
refactor : change ErrorCode to CustomerErrorCode

IM-GYURI committed:
refactor : change ErrorCode to SellerErrorCode

IM-GYURI committed:
refactor : change enum to interface

IM-GYURI committed:
refactor : change CustomException to CommonCustomException

IM-GYURI committed:
refactor : change CustomException to SellerCustomException

IM-GYURI committed:
refactor : change ErrorCode to SellerErrorCode

IM-GYURI committed:
refactor : change ErrorCode to CommonErrorCode

IM-GYURI committed:
refactor : change CustomException to SellerCustomException

IM-GYURI committed:
refactor : change ErrorCode to SellerErrorCode

IM-GYURI committed:
refactor : CommonCustomException implements CustomException

IM-GYURI committed:
refactor : CustomerCustomException implements CustomException

IM-GYURI committed:
refactor : SellerCustomException implements CustomException

IM-GYURI committed:
refactor : CommonErrorCode implements ErrorCode

IM-GYURI committed:
refactor : CustomerErrorCode implements ErrorCode

IM-GYURI committed:
refactor : SellerErrorCode implements ErrorCode
